app/src/main/java/ch/epfl/sdp/game/Server.java

```diff
@@ -97,6 +100,19 @@ public void update() {
         counter++;
     }
 
+    private void checkIfWon() {
+        boolean isWinner = true;
+        for (Player player : playerManager.getPlayers()) {
+            if (player.getHealthPoints() > 0 && player != playerManager.getCurrentUser()) {
+                isWinner = false;
+                break;
+            }
+        }
+        if (isWinner) {
+            endGame.run();
+        }
+    }
+
```

Comment:
So, you don't win iff other players are alive? Then if everyone died, do everyone wins?

Reply:
This is not possible, because the server will win before everyone is dead (and if the server dies, then the game ends). Mabye if they die on the exact same time, but that is very unlikely to happen and then everyone should lose, as the game over activity looks at the healthpoints